src/server/tiering/external_alloc.cc

```diff
@@ -386,6 +399,11 @@ void ExternalAllocator::Free(size_t offset, size_t sz) {
   DCHECK_EQ(page->available, page->free_blocks.count());
   if (page->available == blocks_num) {
     FreePage(page, seg, block_size);
+  } else if (page->available == 1) {
+    uint8_t bin_idx = ToBinIdx(sz);
```

Comment:
should be equal to `page->block_size_bin` i think. also maybe `block_size_bin` is not the best name, maybe `bin_idx` is a better name?